department-of-veterans-affairs/vets-json-schema
# Use profileAddress for Pensions addresses

# New schema
The profileAddress is now used in vets-website with the addressUI.

[profileAddress in common definitions](https://github.com/department-of-veterans-affairs/vets-json-schema/blob/d78aabba46e0a6b6b30bb9fd08f22321a43cd10d/src/common/definitions.js#L267)
[profileAddress being used in vets-website](https://github.com/department-of-veterans-affairs/vets-website/blob/d12833e07bb4f2f94de952e866ddd18bbcf51297/src/platform/forms-system/src/js/web-component-patterns/addressPattern.jsx#L600)

[Issue caused by mismatch between vets-website and vets-json-schema schemas](https://github.com/department-of-veterans-affairs/va.gov-team/issues/86037)

## Testing bug replication locally

- Fill out a pension claim. Have at least one dependent with a street that is longer than 51 and shorter than 100 characters. You will see no error on the front end, but when you try to submit, it will fail.
- Checkout this branch in `vets-json-schema` locally
- Update your vets-api Gemfile to say `gem 'vets_json_schema', path: '/path/to/local/vets-json-schema'` and run `bundle install`
- Attempt to submit the same claim from before and it will succeed.

## Pull Requests to update the schema in related repositories
_After you've merged your changes to vets-json-schema you'll need to make PR's to vets-website and vets-api. Please link them here._

- https://github.com/department-of-veterans-affairs/vets-api/pull/
- https://github.com/department-of-veterans-affairs/vets-website/pull/
